crates/router/src/core/payments/session_operation.rs

```diff
@@ -0,0 +1,189 @@
+pub use hyperswitch_domain_models::{
+    mandates::{CustomerAcceptance, MandateData},
+    payment_address::PaymentAddress,
+    payments::HeaderPayload,
+    router_data::{PaymentMethodToken, RouterData},
+    router_request_types::CustomerDetails,
+};
+use router_env::{instrument, tracing};
+
+use crate::{
+    core::{
+        errors::{self, utils::StorageErrorExt, RouterResult},
+        payments::{
+            call_multiple_connectors_service,
+            flows::{ConstructFlowSpecificData, Feature},
+            operations,
+            operations::{BoxedOperation, Operation},
+            transformers, OperationSessionGetters, OperationSessionSetters,
+        },
+    },
+    errors::RouterResponse,
+    routes::{app::ReqState, SessionState},
+    services,
+    types::{self as router_types, api, domain},
+};
+
+#[cfg(feature = "v2")]
+#[allow(clippy::too_many_arguments)]
+pub async fn payments_session_core<F, Res, Req, Op, FData, D>(
+    state: SessionState,
+    req_state: ReqState,
+    merchant_account: domain::MerchantAccount,
+    profile: domain::Profile,
+    key_store: domain::MerchantKeyStore,
+    operation: Op,
+    req: Req,
+    payment_id: id_type::GlobalPaymentId,
+    call_connector_action: CallConnectorAction,
+    header_payload: HeaderPayload,
+) -> RouterResponse<Res>
+where
+    F: Send + Clone + Sync,
+    Req: Send + Sync,
+    FData: Send + Sync + Clone,
+    Op: Operation<F, Req, Data = D> + Send + Sync + Clone,
+    Req: Debug,
+    D: OperationSessionGetters<F> + OperationSessionSetters<F> + Send + Sync + Clone,
+    Res: transformers::ToResponse<F, D, Op>,
+    // To create connector flow specific interface data
+    D: ConstructFlowSpecificData<F, FData, router_types::PaymentsResponseData>,
+    RouterData<F, FData, router_types::PaymentsResponseData>: Feature<F, FData>,
+
+    // To construct connector flow specific api
+    dyn api::Connector:
+        services::api::ConnectorIntegration<F, FData, router_types::PaymentsResponseData>,
+{
+    let (payment_data, _req, customer, connector_http_status_code, external_latency) =
+        payments_session_operation_core::<_, _, _, _, _>(
+            &state,
+            req_state,
+            merchant_account.clone(),
+            key_store,
+            profile,
+            operation.clone(),
+            req,
+            payment_id,
+            call_connector_action,
+            header_payload.clone(),
+        )
+        .await?;
+
+    Res::generate_response(
+        payment_data,
+        customer,
+        &state.base_url,
+        operation,
+        &state.conf.connector_request_reference_id_config,
+        connector_http_status_code,
+        external_latency,
+        header_payload.x_hs_latency,
+        &merchant_account,
+    )
+}
+
+#[allow(clippy::too_many_arguments, clippy::type_complexity)]
+#[instrument(skip_all, fields(payment_id, merchant_id))]
+pub async fn payments_session_operation_core<F, Req, Op, FData, D>(
```

Comment:
why do we need a separate function for this, can't we reuse `payments_operation_core`?

Reply:
`fn payments_operation_core` has this trait requirement `PaymentResponse: Operation<F, FData, Data = D>,` which is not needed for session token flow.

Reply:
we can use intent_core for this right?